scipy/_build_utils/src/wrap_dummy_g77_abi.c

```diff
@@ -39,69 +39,92 @@ typedef double _Complex double_complex;
 typedef float _Complex float_complex;
 #endif
 
-float_complex BLAS_FUNC(cdotc)(CBLAS_INT *n, float_complex *cx, CBLAS_INT *incx, \
-    float_complex *cy, CBLAS_INT *incy);
-float_complex F_FUNC(wcdotc,WCDOTC)(CBLAS_INT *n, float_complex *cx, \
-        CBLAS_INT *incx, float_complex *cy, CBLAS_INT *incy){
+/* Let linker strip unused symbols from final binaries */
+#if __GNUC__ >= 4
+#define HIDDEN_SYMBOL  __attribute__ ((visibility ("hidden")))
+#else
+#define HIDDEN_SYMBOL
+#endif
```

Comment:
Clang understands `__attribute__ ((visibility ("hidden")))`, though just from the condition, it's not obvious that this branch will also be taken there.

As it happens, clang will also [populate](https://clang.llvm.org/docs/UsersManual.html#cmdoption-fgnuc-version) `__GNUC__`, and while the default value is not documented (from what I saw), it just so happens to be `4`, see:
* https://github.com/llvm/llvm-project/blob/llvmorg-18.1.3/clang/lib/Driver/ToolChains/Clang.cpp#L6892
* https://github.com/llvm/llvm-project/blob/llvmorg-18.1.3/clang/lib/Frontend/CompilerInvocation.cpp#L3838
* https://github.com/llvm/llvm-project/blob/llvmorg-18.1.3/clang/lib/Frontend/InitPreprocessor.cpp#L794

In other words, the `4` here is load-bearing for clang support, and we cannot increment it to our lowest supported GCC version, for example. We definitely need to document this.

Reply:
Oh wow, I had no idea Clang defined such a low value for `__GNUC__`. I copied these defines from [the GCC wiki](https://gcc.gnu.org/wiki/Visibility) and just assumed things would be fine since they worked on my local machine. I moved this into `npy_cblas.h` as you suggested and added a comment about which compilers this targets and why it exists.